### PR TITLE
Update Payment Flows for Barcode Scanning

### PR DESCRIPTION
**Pull request by heymatthew**
This was missleading one of our integrators so it's been updated to reflect how you should interact with our APIs.

image

### CODE REVIEW

File: src/guides/payment-flows.md [hunk 1]
```diff
@@ -55,26 +55,27 @@ barcode or QR code displayed on the customer's smart device.
...
+   parameter.
 3. Customer device [fetches the payment request] created with the Patron Code and completes
    payment as normal.
 
 ### Static Patron Barcode
 
-Variant on Dynamic Patron Code, if you've setup long lived static Patron Codes by integrating with
-Centrapay, then you can use these to pay on behalf of a Customer. The customer may be asked for
-approval before this goes through.
+This is a Variant on Dynamic Patron Code. If you've setup long lived Centrapay
```
Comment (laurencemcknight) [inline src/guides/payment-flows.md:67]:
```suggestion
This is a variant on Dynamic Patron Code. If you've setup a long lived Centrapay
```

**Review by laurencemcknight — APPROVED** [review]

File: src/guides/payment-flows.md [hunk 2]
```diff
@@ -55,26 +55,27 @@ barcode or QR code displayed on the customer's smart device.
...
-   information on the Customer and then [creates a Payment Request][] via Centrapay API,
-   with the Patron Code as a parameter.
-3. Conditionally, the Customer is sent a text message to confirm payment.
-4. The Customer is sent an email with a summary of the payment.
-5. Merchant polls [Polls for the Payment Request][] created with the Patron Code checking the status
-   changes to "paid". Merchant completes payment as normal.
+1. Customer presents a Centrapay Enabled Barcode and it is scanned by the Merchant.
+2. Merchant checks the barcode for support using the [Scanned Codes][] endpoints.
+3. Merchant then [creates a Payment Request][] using the barcode as a parameter.
+4. Merchant [polls for the Payment Request][] created with the Patron Code checking the status.
```
Comment (Ray2126) [inline src/guides/payment-flows.md:74]:
```suggestion
4. Merchant [polls for the Payment Request][] created with the barcode checking the status.
```

File: src/guides/payment-flows.md [hunk 3]
```diff
@@ -55,26 +55,27 @@ barcode or QR code displayed on the customer's smart device.
...
-1. The Customer presents their [Patron Code][] barcode.
-2. Merchant scans the barcode, optionally [retrieves the Patron Code][] for additional
-   information on the Customer and then [creates a Payment Request][] via Centrapay API,
-   with the Patron Code as a parameter.
-3. Conditionally, the Customer is sent a text message to confirm payment.
-4. The Customer is sent an email with a summary of the payment.
-5. Merchant polls [Polls for the Payment Request][] created with the Patron Code checking the status
-   changes to "paid". Merchant completes payment as normal.
+1. Customer presents a Centrapay Enabled Barcode and it is scanned by the Merchant.
+2. Merchant checks the barcode for support using the [Scanned Codes][] endpoints.
```
Comment (Ray2126) [inline src/guides/payment-flows.md:72]: Shall we be specific here and point them to the decode endpoint?

File: src/images/static-patron-code-flow.svg [hunk 4]
```diff
@@ -1 +1 @@
...
...000006 697.8650160579999 L 17.591757400000006 543.233468512 Z" stroke-miterlimit="10" stroke-width="2.5131082" stroke-dasharray=""/><path fill="white" stroke="black" paint-order="fill stroke markers" d=" M 17.591757400000006 543.233468512 L 17.591757400000006 564.3435773919999 L 117.30860502470703 564.3435773919999 L 127.86365946470704 553.788522952 L 127.86365946470704 543.233468512 L 17.591757400000006 543.233468512" stroke-miterlimit="10" stroke-width="2.5131082" stroke-dasharray=""/><text fill="black" stroke="none" font-family="sans-serif" font-size="8.8pt" font-style="normal" font-weight="bold" text-decoration="normal" x="35.183514800000005" y="557.306874432" text-anchor="start" dominant-baseline="alphabetic" xml:space="preserve">Conditional</text><g><rect fill="white" stroke="none" x="142.81665325470703" y="544.640809104" width="173.5775302717578" height="18.295427696"/></g><text fill="black" stroke="none" font-family="sans-serif" font-size="8.8pt" font-style="normal" font-weight="bold" text-decoration="normal" x="145.45541686470705" y="557.306874432" text-anchor="start" dominant-baseline="alphabetic" xml:space="preserve">[Centrapay Triggers ID Check]</text></g></g></g></svg>
```
Comment (Ray2126) [inline src/images/static-patron-code-flow.svg:1]: The create payment request call in this diagram has a `barcodeType` parameter. Does this exist/will it exist in the future?

File: src/guides/payment-flows.md [hunk 5]
```diff
@@ -55,26 +55,27 @@ barcode or QR code displayed on the customer's smart device.
 
 1. Customer generates a short-lived [Patron Code][] via Centrapay API and displays
    it as a barcode on their smart device
-2. Merchant scans the barcode, optionally [retrieves the Patron Code][] for additional
-   information on the customer and then [creates a Payment Request][] via Centrapay API,
-   with the Patron Code as a parameter.
+2. Merchant scans the barcode, optionally checks for support using the
```
Comment (Ray2126) [inline src/guides/payment-flows.md:58]: Ditto, can we point them to the specific endpoint they should use to check for support?

File: src/guides/payment-flows.md [hunk 6]
```diff
@@ -55,26 +55,27 @@ barcode or QR code displayed on the customer's smart device.
...
+   [Scanned Codes][] endpoints for additional information on the customer, and then
+   [creates a Payment Request][] via Centrapay API with the Patron Code as a
```
Comment (Ray2126) [inline src/guides/payment-flows.md:60]:
```suggestion
   [creates a Payment Request][] via Centrapay API with the barcode as a
```

File: src/guides/payment-flows.md [hunk 7]
```diff
@@ -55,26 +55,27 @@ barcode or QR code displayed on the customer's smart device.
 
 1. Customer generates a short-lived [Patron Code][] via Centrapay API and displays
    it as a barcode on their smart device
-2. Merchant scans the barcode, optionally [retrieves the Patron Code][] for additional
-   information on the customer and then [creates a Payment Request][] via Centrapay API,
-   with the Patron Code as a parameter.
+2. Merchant scans the barcode, optionally checks for support using the
+   [Scanned Codes][] endpoints for additional information on the customer, and then
```
Comment (Ray2126) [inline src/guides/payment-flows.md:59]: I'm not sure what is meant by `additional information on the customer` here. We do not get information about the customer from the Scanned code endpoints. Should this mention something about discounts?

**Review by Ray2126 — CHANGES_REQUESTED** [review]

File: src/guides/payment-flows.md (hunk 7 above)
Comment (ncjones) [inline src/guides/payment-flows.md:59]:
```suggestion
   [Scanned Codes][] endpoints for additional information on the patron, and then
```

**Review by ncjones — APPROVED** [review]

**Pull request merged**